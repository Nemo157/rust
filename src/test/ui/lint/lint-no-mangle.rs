#![allow(dead_code)]
#![deny(no_mangle)]

struct Bar;

#[allow(no_mangle)]
mod allowed_no_mangle {
    #[no_mangle] fn allowed() {}
}

macro_rules! no_mangle_in_macro {
    () => {
        #[no_mangle] fn foo() {} //~ ERROR: declaration of a `no_mangle` function
        #[no_mangle] static FOO: u32 = 5; //~ ERROR: declaration of a `no_mangle` static
    }
}

#[no_mangle] fn foo() {} //~ ERROR: declaration of a `no_mangle` function
#[no_mangle] static FOO: u32 = 5; //~ ERROR: declaration of a `no_mangle` static

fn main() {
    no_mangle_in_macro!();
}
