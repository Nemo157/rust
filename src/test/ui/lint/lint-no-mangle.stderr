error: declaration of a `no_mangle` function
  --> $DIR/lint-no-mangle.rs:18:14
   |
LL | #[no_mangle] fn foo() {}
   |              ^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/lint-no-mangle.rs:2:9
   |
LL | #![deny(no_mangle)]
   |         ^^^^^^^^^

error: declaration of a `no_mangle` static
  --> $DIR/lint-no-mangle.rs:19:14
   |
LL | #[no_mangle] static FOO: u32 = 5;
   |              ^^^^^^^^^^^^^^^^^^^^

error: declaration of a `no_mangle` function
  --> $DIR/lint-no-mangle.rs:13:22
   |
LL |         #[no_mangle] fn foo() {}
   |                      ^^^^^^^^^^^
...
LL |     no_mangle_in_macro!();
   |     ---------------------- in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: declaration of a `no_mangle` static
  --> $DIR/lint-no-mangle.rs:14:22
   |
LL |         #[no_mangle] static FOO: u32 = 5;
   |                      ^^^^^^^^^^^^^^^^^^^^
...
LL |     no_mangle_in_macro!();
   |     ---------------------- in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 4 previous errors

